File: homedir/.scripts/rename.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

func main() {
	// Get the current directory
	dir, err := os.Getwd()
	if err != nil {
		fmt.Println("Error getting current directory:", err)
		return
	}

	// Read all files in the current directory
	files, err := ioutil.ReadDir(dir)
	if err != nil {
		fmt.Println("Error reading files in the current directory:", err)
		return
	}

	// Loop over each file
	for _, file := range files {
		// Check if the file is not a directory and has a .mp4 extension
		if !file.IsDir() && filepath.Ext(file.Name()) == ".mp4" {
			// Get the creation time of the file
			createdTime := file.ModTime()

			// Generate the new name with the ISO format of the created time
			newName := "NAME-" + createdTime.Format("2006-01-02") + filepath.Ext(file.Name())

			// Rename the file
			err := os.Rename(file.Name(), newName)
			if err != nil {
				fmt.Printf("Error renaming file %s to %s: %v\n", file.Name(), newName, err)
			} else {
				fmt.Printf("Renamed %s to %s\n", file.Name(), newName)
			}
		}
	}
}

File: homedir/.scripts/rename.py
#!/usr/bin/env python3

import argparse
import datetime
import os

# Parse command line arguments
def parse_args():
    parser = argparse.ArgumentParser(description="Rename files in directoy by created date.")
    parser.add_argument('baseName', type=str, help='Base name for every file')
    return parser.parse_args()

def main():
    # Parse command line arguments
    args = parse_args()
    baseName = args.baseName
    # Get the current directory
    dir = os.getcwd()

    try:
        # Read all files in the current directory
        files = os.listdir(dir)
    except OSError as e:
        print("Error reading files in the current directory:", e)
        return

    # Loop over each file
    for file in files:
        file_path = os.path.join(dir, file)
        
        # Check if the file is not a directory and has a .mp4 extension
        if os.path.isfile(file_path) and file.endswith(".mp4"):
            try:
                # Get the creation time of the file
                created_time = datetime.datetime.fromtimestamp(os.path.getmtime(file_path))

                # Generate the new name with the ISO format of the created time
                new_name = baseName + "-" + created_time.strftime("%Y-%m-%d") + ".mp4"
                new_file_path = os.path.join(dir, new_name)

                # Rename the file
                os.rename(file_path, new_file_path)
                print(f"Renamed {file} to {new_name}")
            except OSError as e:
                print(f"Error renaming file {file} to {new_name}: {e}")

if __name__ == "__main__":
    main()